Source/Common/ValuesJsonConverter.cs

```diff
@@ -6,104 +6,81 @@ namespace Schema.NET
     /// Converts a <see cref="IValues"/> object to and from JSON.
     /// </summary>
     /// <seealso cref="JsonConverter" />
-    public class ValuesJsonConverter : JsonConverter
+    public class ValuesJsonConverter : JsonConverter<IValues>
     {
-        private const string HttpSchemaOrgUrl = "http://schema.org/";
-        private const int HttpSchemaOrgLength = 18; // equivalent to "http://schema.org/".Length
-        private const string HttpsSchemaOrgUrl = "https://schema.org/";
-        private const int HttpsSchemaOrgLength = 19; // equivalent to "https://schema.org/".Length
-
-        private static readonly TypeInfo ThingInterfaceTypeInfo = typeof(IThing).GetTypeInfo();
         private static readonly Dictionary<string, Type> BuiltInThingTypeLookup = new(StringComparer.Ordinal);
 
         static ValuesJsonConverter()
         {
-            var thisAssembly = ThingInterfaceTypeInfo.Assembly;
+            var thisAssembly = typeof(IThing).Assembly;
             foreach (var type in thisAssembly.ExportedTypes)
             {
-                var typeInfo = type.GetTypeInfo();
-                if (typeInfo.IsClass && ThingInterfaceTypeInfo.IsAssignableFrom(typeInfo))
+                if (type.IsClass && typeof(IThing).IsAssignableFrom(type))
                 {
                     BuiltInThingTypeLookup.Add(type.Name, type);
                 }
             }
         }
 
-        /// <summary>
-        /// Determines whether this instance can convert the specified object type.
-        /// </summary>
-        /// <param name="objectType">Type of the object.</param>
-        /// <returns>
-        /// <c>true</c> if this instance can convert the specified object type; otherwise, <c>false</c>.
-        /// </returns>
-        public override bool CanConvert(Type objectType) => objectType == typeof(IValues);
+        /// <inheritdoc />
+        public override bool CanConvert(Type typeToConvert) => typeof(IValues).IsAssignableFrom(typeToConvert);
 
         /// <summary>
         /// Reads the JSON representation of the object.
         /// </summary>
-        /// <param name="reader">The <see cref="JsonReader"/> to read from.</param>
-        /// <param name="objectType">Type of the object.</param>
-        /// <param name="existingValue">The existing value of object being read.</param>
-        /// <param name="serializer">The calling serializer.</param>
+        /// <param name="reader">The <see cref="Utf8JsonReader"/> to read from.</param>
+        /// <param name="typeToConvert">Type of the object.</param>
+        /// <param name="options">The serializer options.</param>
         /// <returns>The object value.</returns>
-        public override object? ReadJson(
-            JsonReader reader,
-            Type objectType,
-            object? existingValue,
-            JsonSerializer serializer)
+        public override IValues? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
```

Comment:
We should check `reader` for `null` here.

Reply:
Reader can't be null, it is a `struct`. Not sure if we should be checking then if its `default` then but my understanding is that we can trust it to be OK.